scripts/install/common/error-handling.test.sh

```
@@ -0,0 +1,346 @@
+    test_start "run_idempotent does not mark failed tasks as done"
+    
+    local tmp_home=$(mktemp -d)
+    
+    # Run a failing command
+    # We expect it to fail, so we ignore the exit code of bash -c with || true
+    TMPDIR="$tmp_home" bash -c "source '$LIB_PATH'; \
+        run_idempotent 'step_fail' false" > /dev/null 2>&1 || true
+    
+    # Check if the marker file exists (it should NOT)
+    # The user-specific directory might vary, so we find it
+    local state_dirs=$(find "$tmp_home" -name "talawa-install-state*")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)